dashboard/app/models/authentication_option.rb

```diff
@@ -48,25 +48,43 @@ class AuthenticationOption < ApplicationRecord
     TWITTER = 'twitter',
     WINDOWS_LIVE = 'windowslive',
     MICROSOFT = 'microsoft_v2_auth',
-  ]
+  ].freeze
 
+  CREDENTIAL_TYPES = [
+    EMAIL = 'email',
+    OAUTH_CREDENTIAL_TYPES,
+  ].flatten.freeze
+
+  # "untrusted" emails are a somewhat subtle concept.
+  #
+  # They specifically refer to emails we receive from a provider that
+  #
+  # A) Does not themselves enforce uniqueness and/or
+  # B) Does not allow the user to change the email they have been assigned.
```

Comment:
A subtle thing: You wrote "and/or" but I think it should just be "and."  Other OAuth providers may not make it easy for the user to change the email they have been assigned (or more saliently, which email is sent to our system) but if they themselves enforce uniqueness we aren't listing them as "untrusted."